django/db/models/sql/compiler.py

```diff
@@ -1911,18 +1910,6 @@ def execute_sql(self, returning_fields=None):
                     )
                 ]
                 cols = [field.get_col(opts.db_table) for field in self.returning_fields]
-            elif isinstance(opts.pk, CompositePrimaryKey):
-                returning_field = returning_fields[0]
```

Comment:
This could be wrong if the `returning_fields` included `GeneratedField` members as nothing guarantees that `returning_fields[0]` will be the `AutoField` that must be passed to `last_insert_id`.